packages/backend/src/telemetry/service.ts

```diff
@@ -0,0 +1,145 @@
+import { Counter, MetricOptions, metrics } from '@opentelemetry/api'
+import { OTLPMetricExporter } from '@opentelemetry/exporter-metrics-otlp-grpc'
+import { Resource } from '@opentelemetry/resources'
+import {
+  MeterProvider,
+  PeriodicExportingMetricReader
+} from '@opentelemetry/sdk-metrics'
+
+import { ConvertError, RatesService, isConvertError } from '../rates/service'
+import { ConvertOptions } from '../rates/util'
+import { BaseService } from '../shared/baseService'
+
+export interface TelemetryService {
+  shutdown(): void
+  getOrCreateMetric(name: string, options?: MetricOptions): Counter
+  getInstanceName(): string | undefined
+  getBaseAssetCode(): string
+  getBaseScale(): number
+  convertAmount(
+    convertOptions: Omit<ConvertOptions, 'exchangeRate' | 'destinationAsset'>
+  ): Promise<bigint | ConvertError>
+}
+
+interface TelemetryServiceDependencies extends BaseService {
+  instanceName: string
+  collectorUrls: string[]
+  exportIntervalMillis?: number
+  aseRatesService: RatesService
+  fallbackRatesService: RatesService
+  baseAssetCode: string
+  baseScale: number
+}
+
+const METER_NAME = 'Rafiki'
+const SERVICE_NAME = 'RAFIKI_NETWORK'
+
+export function createTelemetryService(
+  deps: TelemetryServiceDependencies
+): TelemetryService {
+  return new TelemetryServiceImpl(deps)
+}
+
+class TelemetryServiceImpl implements TelemetryService {
+  private instanceName: string
+  private meterProvider?: MeterProvider
+  private fallbackRatesService: RatesService
+  private aseRatesService: RatesService
+
+  private counters = new Map()
+  constructor(private deps: TelemetryServiceDependencies) {
+    // debug logger:
+    // diag.setLogger(new DiagConsoleLogger(), DiagLogLevel.DEBUG)
```

Comment:
Do we need to uncomment these?

Reply:
no, I'm leaving the comment there just for when debugging is needed on local development setup.
keeping the debug logger on at all times would also log on every 15 seconds, even if new metrics are not there.
So i'd rather just keep them off.